src/snowflake/cli/plugins/nativeapp/manager.py

```diff
@@ -571,3 +575,39 @@ def deploy(
             )
 
         return diff
+
+    def validate(self):
+        """Call system$validate_native_app_setup() to validate deployed Native App setup script."""
+        cc.step(f"Validating Native App setup script.")
+        stage_name = StageManager.get_standard_stage_prefix(self.stage_fqn)
+        try:
+            cursor = self._execute_query(
+                f"call system$validate_native_app_setup('{stage_name}')"
+            )
+        except ProgrammingError as err:
+            if "does not exist or not authorized" in err.msg:
+                raise ApplicationPackageDoesNotExistError(self.package_name)
+            generic_sql_error_handler(err)
+        else:
+            if not cursor.rowcount:
+                raise SnowflakeSQLExecutionError()
+            result_data = json.loads(cursor.fetchone()[0])
+
+            # First print warnings, regardless of the outcome of validation
+            for warning in result_data.get("warnings", []):
+                cc.warning(_validation_item_to_str(warning))
+
+            # Then raise an exception if validation failed
+            if result_data["status"] == "FAIL":
```

Comment:
Is `status` always an existing key in `result_data`? or can we get a `KeyError` here?

Reply:
It's always set, and defaults to SUCCESS: https://github.com/snowflakedb/snowflake/blob/d88930c088fb783c3b3a23acd4f6f1a6d776628c/GlobalServices/src/main/java/com/snowflake/metadata/dictionary/NativeAppsValidationResult.java#L130. It gets set to FAIL on the first error: https://github.com/snowflakedb/snowflake/blob/d88930c088fb783c3b3a23acd4f6f1a6d776628c/GlobalServices/src/main/java/com/snowflake/metadata/dictionary/NativeAppsValidationResult.java#L156-L158

Reply:
yeah it's always set in the response. It defaults to SUCCESS and gets flipped to FAIL on the first recorded error